go.mod

```diff
@@ -1,5 +1,5 @@
 module github.com/erigontech/mdbx-go
 
-go 1.15
+go 1.22
```

Comment:
It does not affect "which compiler will be used to compile code" (go.mod had `toolchain` directive for this).
I would advise increase it only if it gives us some value. 
Erigon also now supports 1.21

Reply:
see also `.github/workflows/test.yml`

Reply:
lookup at test.yml, they used go 1.20 changed go mod to the same version